### DISCUSSION TITLE
Error: Could not find Chromium (rev. 1056772)

### DISCUSSION THREAD

**Discussion by moxesh-gandhi** [spatie/browsershot#686] [Bugs]
The command "PATH=$PATH:/usr/local/bin NODE_PATH=`npm root -g` node '/var/www/html/browsershot/vendor/spatie/browsershot/src/../bin/browser.js' '{"url":"https:\/\/google.com","action":"screenshot","options":{"type":"png","path":"storage\/googlescreenshot.jpg","args":[],"viewport":{"width":3840,"height":2160},"landscape":true,"waitUntil":"networkidle0"}}'" failed. Exit Code: 1(General error) Working directory: /var/www/html/browsershot/public Output: ================ Error Output: ================ ┌──────────────────────────────────────────────────────┐ │ npm update check failed │ │ Try running with sudo or get access │ │ to the local update config store via │ │ sudo chown -R $USER:$(id -gn $USER) /var/www/.config │ └──────────────────────────────────────────────────────┘ Error: Could not find Chromium (rev. 1056772). This can occur if either 1. you did not perform an installation before running the script (e.g. `npm install`) or 2. your cache path is incorrectly configured (which is: /var/www/.cache/puppeteer). For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration. at ChromeLauncher.resolveExecutablePath (/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ProductLauncher.js:120:27) at ChromeLauncher.executablePath (/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:166:25) at ChromeLauncher.launch (/node_modules/puppeteer-core/lib/cjs/puppeteer/node/ChromeLauncher.js:70:37) at async callChrome (/var/www/html/browsershot/vendor/spatie/browsershot/bin/browser.js:69:23)

**Comment by kokoshneta** [comment]
Not much info to go on there… just pasting an error message on its own (especially without even bothering to format it!) isn’t enough for anyone to help you.

However, I get the same error message when I try to run Browsershot on my server (Windows 2019), so I’m in the same boat as you. Your error message seems to indicate that the cache path given is not the expected one; the same is true for me.

I have Node and NPM installed at the latest versions, and Puppeteer itself (version 19.2.2) installs with no issues. Browsershot is required in Composer and is called correctly.

According to the [configuration page](https://pptr.dev/guides/configuration/#changing-the-default-cache-directory), the default location of the Chromium browser that is downloaded and installed with Puppeteer since version 19.0.0 is `~/.cache/puppeteer`. Indeed, when I look in the current user’s folder on the server, this folder exists and contains a subfolder `chrome/win64-1056772/chrome-win`, in which there is a `chrome.exe` executable.

But I get roughly the same error message as you:

```text
Error: Could not find Chromium (rev. 1056772). This can occur if either

1. you did not perform an installation before running the script (e.g. `npm install`) or
2. your cache path is incorrectly configured (which is: C:\Windows\system32\config\systemprofile\.cache\puppeteer).

For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration.

	at ChromeLauncher.resolveExecutablePath (C:\inetpub\vhosts\my_project\app\node_modules\puppeteer-core\lib\cjs\puppeteer\node\ProductLauncher.js:120:27)

	at ChromeLauncher.executablePath (C:\inetpub\vhosts\my_project\app\node_modules\puppeteer-core\lib\cjs\puppeteer\node\ChromeLauncher.js:166:25)

	at ChromeLauncher.launch (C:\inetpub\vhosts\my_project\app\node_modules\puppeteer-core\lib\cjs\puppeteer\node\ChromeLauncher.js:70:37)

	at async callChrome (C:\inetpub\vhosts\my_project\app\vendor\spatie\browsershot\bin\browser.js:84:23)
```

My cache path is given here as `C:\Windows\system32\config\systemprofile\.cache\puppeteer`, which seems functionally comparable to your `/var/www/.cache/puppeteer`. There is no `.cache` folder in my `systemprofile` folder. I’m not sure if the path in the error message is generated dynamically, or if it’s just a static path that hasn’t been updated since the default location was changed in version 19.0.0. I’ve tried copying the entire cache folder into the `systemprofile` folder, just in case, but this had no effect.

As [defined in the Puppeteer configuration docs](https://pptr.dev/api/puppeteer.configuration.cachedirectory), I’ve tried adding a `.puppeteerrc.cjs` file in the project root, as well as defining an environmental variable `PUPPETEER_CACHE_DIR`, and then reinstalling Puppeteer. The config file definitely gets picked up when reinstalling Puppeteer, because I made a typo in it first time around, which caused the installer to fail completely.

I’ve also tried calling Browsershot with [a custom Chromium executable path](https://spatie.be/docs/browsershot/v2/requirements#content-custom-chromechromium-executable-path), copy-pasting various permutations of the absolute and relative paths to the executable or its containing folder.

All to no effect. Puppeteer still cannot find Chromium.

**Comment by kokoshneta** [comment]
Based on [this previous discussion](https://github.com/spatie/browsershot/discussions/681), I installed Puppeteer 17.1.3 instead, and it immediately found Chromium..

**Reply by ardyjoe** [reply]
works for me also
Thanks!

**Comment by MCKLtech** [comment]
If you are using Heroku @moxesh-gandhi, I had the same issue and solved it by downgrading:

```npm install puppeteer@~17.1.3```

**Reply by davidcalhoun** [reply]
Thank you, this worked for me as well using Yarn.  I created a new Puppeteer bug here: https://github.com/puppeteer/puppeteer/issues/9304

**Reply by rileysiemens** [reply]
this also worked for me

**Reply by ardyjoe** [reply]
works for me also
Thanks!

**Reply by livingos** [reply]
This worked for me too.
